docs/USAGE.md

```diff
@@ -0,0 +1,106 @@
+# Usage
+
+
+## Basic library usage
+***
+
+The API is in late alpha testing, and has not yet reached a beta (pre-release) stage.
+
+Here is an example of usage (circa v1.4)
+
+```python3
+import multicast as multicast
+from multiprocessing import Process as Process
+
+# set up some stuff
+_fixture_PORT_arg = int(59595)
+_fixture_mcast_GRP_arg = """224.0.0.1"""  # only use dotted notation for multicast group addresses
+_fixture_host_BIND_arg
+_fixture_HEAR_args = [
+	"""--port""", _fixture_PORT_arg,
+	"""--join-mcast-groups""", _fixture_mcast_GRP_arg,
+	"""--bind-group""", _fixture_mcast_GRP_arg
+]
+
+# spawn a listening proc
+
+def inputHandle():
+	test_RCEV = multicast.recv.McastRECV()
+	buffer_string = str("""""")
+	buffer_string += test_RCEV._hearstep([_fixture_mcast_GRP_arg], _fixture_PORT_arg, _fixture_host_BIND_arg, _fixture_mcast_GRP_arg)
```

Comment:
🙉 Fair point.
* Read "multicast/__main__.py" focusing on the use of the "McastRECV" class.
* Read "multicast/__init__.py" focusing on the usage of the "doStep" in the abstract "mtool" class.
* Read "multicast/recv.py" focusing on the "McastRECV" class.
* finally, can you please refactor this example code to use `doStep` instead of `_hearstep`

Reply:
we should use `if result[0]:` for the final condition as that is the convention of return values from `doStep`; `dostep` returns a tuple of (success?, *optional_results)

basicly like this:
```suggestion
	result = test_RCEV.doStep(
        groups=[_fixture_mcast_GRP_arg],
        port=_fixture_PORT_arg,
        iface=_fixture_host_BIND_arg,
        group=_fixture_mcast_GRP_arg,
    )
    if result[0]:
        buffer_string += result[1]
```
or even better:
```suggestion
	(didWork, result) = test_RCEV.doStep(
        groups=[_fixture_mcast_GRP_arg],
        port=_fixture_PORT_arg,
        iface=_fixture_host_BIND_arg,
        group=_fixture_mcast_GRP_arg,
    )
    if didWork:
        buffer_string += result
```

Reply:
LGTM!
```suggestion
	(didWork, result) = test_RCEV.doStep(
		groups=[_fixture_mcast_GRP_arg],
		port=_fixture_PORT_arg,
		iface=_fixture_host_BIND_arg,
		group=_fixture_mcast_GRP_arg,
	)
	if didWork:
		buffer_string += result
```